Sources/Basics/Errors.swift

```diff
@@ -30,6 +30,9 @@ public struct InternalError: Error {
 extension Error {
     public var interpolationDescription: String {
         switch self {
+        // special case because `LocalizedError` conversion will hide the underlying error
+        case let _error as DecodingError:
+            return "\(self)"
         case let _error as LocalizedError:
```

Comment:
So the old code in `error()` had this:

```
} else if let localizedError = error as? LocalizedError {
    message = localizedError.errorDescription ?? localizedError.localizedDescription
}
```

which means we're losing preferring `errorDescription` to `localizedDescription`. It looks like a bunch of our types have an `errorDescription`, so this might be meaningful to preserve.

Reply:
That's a good point — would it be cleaner to convert those over to be spelled `localizedDescription`?  Or might there be implementations of `errorDescription` coming from TSC that might be harder to switch over?

Reply:
It seems like `errorDescription` is a legitimate part of `LocalizedError`: https://developer.apple.com/documentation/foundation/localizederror/2946895-errordescription, so in theory could be coming from types external to SwiftPM/TSC.

Reply:
good point, addressed